### PR TITLE
Add originId to tasks

### PR DESCRIPTION
**Pull request by agluszak**
The main point of this change is to be able to distinguish progress notifications belonging to a specific task tree in the case of multiple concurrent tasks triggered by different requests (or no request).

### CODE REVIEW

**Review by jastice — APPROVED** [review]
From our discussion for context: the main point of this change is to be able to distinguish progress notifications belonging to a specific task tree in the case of multiple concurrent tasks triggered by different requests (or no request).

Please add the context to the commit message.

Let's give this until 29 September to see if somebody wants to comment or has concerns before merging.

**Review by ckipp01 — CHANGES_REQUESTED** [review]
Do you think it'd be possible to also add `originId` in the the roundtrip test suite just to ensure it's appearing and working as expected?

File: README.md [hunk 1]
```diff
@@ -37,3 +37,10 @@ _Past Maintainers_
 
 - Jorge Vicente Cantero - [@jvican](https://github.com/jvican)
 - Ólafur Páll Geirsson - [@olafurpg](https://github.com/olafurpg)
+
+## Protocol changelog
+
+### 2.1.1
+
+- Added an optional `originId` field to `TaskStartParams`, `TaskProgressParams` and `TaskFinishParams`
+  and deprecated it in `CompileReport` and `TestReport` to support BSP clients that need to distinguish between multiple reports for the same target.
```
[the remark this answers is not in the document]
Comment (agluszak) [reply README.md:46]: Not really. Reports live in TaskFinishParams' data field, so if you want to send the originId for report you just put it in the parent field.

**Comment by agluszak** [comment]
> Do you think it'd be possible to also add originId in the the roundtrip test suite just to ensure it's appearing and working as expected?

What do you mean exactly? I thought the roundtrip test suite tests serialization

File: README.md [hunk 2]
```diff
@@ -37,3 +37,10 @@ _Past Maintainers_
 
 - Jorge Vicente Cantero - [@jvican](https://github.com/jvican)
 - Ólafur Páll Geirsson - [@olafurpg](https://github.com/olafurpg)
+
+## Protocol changelog
+
+### 2.1.1
```
Comment (adpi2) [inline README.md:43]: 2.1.1 does not yet exist so I find it weird to mention it here.

Also if we add something to the protocol, should it be a minor version bump? => 2.2.0

```suggestion
### 2.1.1 (future version)
```

Or

```suggestion
### Unreleased
```

**Review by adpi2 — COMMENTED** [review]
Overall it looks good to me.

File: README.md (hunk 2 above)
Comment (agluszak) [reply README.md:43]: I agree, that's why I opened https://github.com/build-server-protocol/build-server-protocol/pull/606

**Review by adpi2 — APPROVED** [review]

**Comment by agluszak** [comment]
@ckipp01 added the test

**Review by ckipp01 — APPROVED** [review]
LGTM!

**Comment by agluszak** [comment]
So since all the interested parties have already accepted I'll merge it today, before the deadline of 29 September. Unless @lefou has something to add?

**Pull request merged**